framework/src/Volo.Abp.MultiTenancy/Volo/Abp/MultiTenancy/TenantConfigurationProvider.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Threading.Tasks;
+using Volo.Abp.DependencyInjection;
+
+namespace Volo.Abp.MultiTenancy
+{
+    public class TenantConfigurationProvider : ITenantConfigurationProvider, ITransientDependency
+    {
+        protected virtual ITenantResolver TenantResolver { get; }
+        protected virtual ITenantStore TenantStore { get; }
+        protected virtual ITenantResolveResultAccessor TenantResolveResultAccessor { get; }
+
+        public TenantConfigurationProvider(
+            ITenantResolver tenantResolver,
+            ITenantStore tenantStore,
+            ITenantResolveResultAccessor tenantResolveResultAccessor)
+        {
+            TenantResolver = tenantResolver;
+            TenantStore = tenantStore;
+            TenantResolveResultAccessor = tenantResolveResultAccessor;
+        }
+
+        public virtual async Task<TenantConfiguration> ResolverAndGet()
+        {
+            var resolveResult = TenantResolver.ResolveTenantIdOrName();
+            TenantResolveResultAccessor.Result = resolveResult;
```

Comment:
The problem of this implementation is that it implicitly sets the `TenantResolveResultAccessor.Result`. This is a side effect since the method is a GET method and overrides a global-like value every time it is called.

Reply:
How about adding a `saveResolveResult ` parameter?
```
Task<TenantConfiguration> GetAsync(bool saveResolveResult = false)
```

Reply:
This would be better, so we can explicitly set it only if we need it.